src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java

```diff
@@ -74,4 +76,27 @@ Page<EventSearchResult> searchEventsFilteringOn(
         LocalDate hearingStartDate,
         LocalDate hearingEndDate,
         Pageable pageable);
-}
+
+    @Query(value = """
+            select distinct evnt.event_id from darts.event evnt
+                inner join (SELECT distinct count(event_id) as eventCount, event_id  FROM
+                darts.event where is_current=true GROUP BY event_id) eventIdCntResult ON evnt.event_id = eventIdCntResult.event_id
+            where evnt.event_id <> 0 and eventIdCntResult.eventCount > 1
+            LIMIT :batchSize
```

Comment:
rather than setting a limit manually ourselves I suggest we let JPA handle that automatically by using the standard JPA approach of doing limit queries using Pageable. We've already used this approach in other repos, and example [here](https://github.com/hmcts/darts-api/blob/78981fd7db29a3608ec8920f81039416f3a7ebc1/src/main/java/uk/gov/hmcts/darts/common/repository/CaseRepository.java#L60)

Reply:
Pagination should work also with native queries in recent versions of spring boot, see [here](https://stackoverflow.com/questions/38349930/spring-data-and-native-query-with-pagination)

Reply:
also I think this query can be replaced with more performant one which replaces the inner join with a HAVING clause:
```
SELECT distinct count(event_id) as eventCount, event_id  FROM
                darts.event where is_current=true 
				AND event_id <> 0 
				GROUP BY event_id
				having count(event_id) > 1
				order by event_id ASC
```